### PR TITLE
Added nix files

### PR DESCRIPTION
**Pull request by guilhermehas**
This pull request can help those who want to build the code from nix.

### CODE REVIEW

**Comment by gallais** [comment]
I don't know anything about nix. Is there any chance we can get a github
action making sure the install script is working?

**Comment by guilhermehas** [comment]
I put a continuous integration of GitHub action and installation instruction.

**Comment by gallais** [comment]
Thanks!

File: result [hunk 1]
```diff
@@ -0,0 +1 @@
+/nix/store/nw0mm1z2afpxv5rgf6w3apxc8pxwgd6h-idris2-0.3.0
```
Comment (peti) [inline result:1]: You probably would not want to check the `result` symlink into the repository, no?

Comment (gallais) [reply result:1]: This was removed in a later commit (#953)

**Pull request merged**